packages/lesswrong/components/votes/ThreeAxisEmojisVoteOnComment.tsx

```diff
@@ -0,0 +1,293 @@
+import React, { FC, MouseEvent, useState, useCallback } from "react";
+import { Components, registerComponent } from "../../lib/vulcan-lib";
+import {
+  CommentVotingComponentProps,
+  getVotingSystemByName,
+} from "../../lib/voting/votingSystems";
+import { useVote, VotingProps } from "./withVote";
+import { usePostsPageContext } from "../posts/PostsPage/PostsPageContext";
+import { useTracking } from "../../lib/analyticsEvents";
+import { useCurrentUser } from "../common/withUser";
+import { useDialog } from "../common/withDialog";
+import { AddEmoji } from "../icons/addEmoji";
+import { eaEmojiPalette, EmojiOption } from "../../lib/voting/eaEmojiPalette";
+import Menu from "@material-ui/core/Menu";
+import classNames from "classnames";
+import { userHasEAEmojiReacts } from "../../lib/betas";
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {
+    display: "flex",
+    gap: "10px",
+  },
+  button: {
+    display: "flex",
+    alignItems: "center",
+    borderRadius: theme.borderRadius.small,
+    background: theme.palette.grey[110],
+    border: `1px solid ${theme.palette.grey[110]}`,
+    color: theme.palette.grey[600],
+    padding: "0 6px",
+    gap: "6px",
+    height: 26,
+    transform: "translateY(-1px)",
+    cursor: "pointer",
+    userSelect: "none",
+    "&:hover": {
+      background: theme.palette.grey[200],
+    },
+  },
+  buttonSelected: {
+    background: theme.palette.primaryAlpha(0.1),
+    border: `1px solid ${theme.palette.primaryAlpha(0.6)}`,
+    color: theme.palette.primary.main,
+  },
+  emojiPreview: {
+    fontSize: "1.4em",
+  },
+  tooltip: {
+    background: theme.palette.grey[800],
+    color: theme.palette.grey[0],
+    transform: "translateY(-8px)",
+    textAlign: "center",
+    maxWidth: 190,
+  },
+  tooltipSecondaryText: {
+    color: theme.palette.grey[400],
+  },
+  tooltipEmoji: {
+    fontSize: "1.4em",
+  },
+  menu: {
+    "& .MuiPaper-root": {
+      transform: "translateY(-8px) !important",
+    },
+    "& .MuiList-padding": {
+      padding: 0,
+    },
+  },
+});
+
+const isEmojiSelected = <T extends VoteableTypeClient>(
+  voteProps: VotingProps<T>,
+  emojiOption: EmojiOption,
+) => Boolean(voteProps.document?.currentUserExtendedVote?.[emojiOption.name]);
+
+const getCurrentReactions = <T extends VoteableTypeClient>(
+  voteProps: VotingProps<T>,
+) => {
+  const extendedScore = voteProps.document?.extendedScore;
+  if (!extendedScore || !Object.keys(extendedScore).length) {
+    return [];
+  }
+
+  const result = [];
+  for (const emojiOption of eaEmojiPalette) {
+    if ((extendedScore[emojiOption.name] ?? 0) > 0) {
+      result.push({
+        emojiOption,
+        score: extendedScore[emojiOption.name],
+      });
+    }
+  }
+  return result;
+}
+
+const joinStringList = (items: string[]) =>
+  items.length > 1
+    ? items.slice(0, -1).join(", ") + " and " + items.at(-1)
```

Comment:
```suggestion
    ? items.slice(0, -1).join(", ") + ", and " + items.at(-1)
```

I forget, but I bet Lizka is on the Oxford comma train. @aarongertler definitely was.

Reply:
I think I'm a fan of oxford comma for UI too. Double checked slack and they do it:
![Screenshot 2023-05-19 at 11 09 01](https://github.com/ForumMagnum/ForumMagnum/assets/13235378/c7bcdb0e-ae3b-41bd-ad0e-9c86a8164267)